Empire/Empire.cpp

```diff
@@ -445,34 +445,28 @@ bool Empire::ProducibleItem(BuildType build_type, int location_id) const {
         throw std::invalid_argument("Empire::ProducibleItem was passed BuildType BT_SHIP with no further parameters, but ship designs are tracked by number");
     
     if (build_type == BT_BUILDING)
-        throw std::invalid_argument("Empire::ProducibleItem was passed BuildType BT_BUILDING with no further parameters, but these types are tracked by name");
+        throw std::invalid_argument("Empire::ProducibleItem was passed BuildType BT_BUILDING with no further parameters, but buildings are tracked by name");
 
     auto build_location = GetUniverseObject(location_id);
     if (!build_location)
         return false;
 
-    if (build_type == BT_STOCKPILE) {
-        Empire* empire = GetEmpire(m_id);
-        if (!empire) {
-            DebugLogger() << "ShipDesign::ProductionLocation: Unable to get pointer to empire " << m_id;
-            return false;
-        }
+    // must own the production location...
+    auto location = GetUniverseObject(location_id);
+    if (!location) {
+        WarnLogger() << "ShipDesign::ProductionLocation unable to get location object with id " << location_id;
+        return false;
+    }
 
-        // must own the production location...
-        auto location = GetUniverseObject(location_id);
-        if (!location) {
-            WarnLogger() << "ShipDesign::ProductionLocation unable to get location object with id " << location_id;
-            return false;
-        }
-        if (!location->OwnedBy(m_id))
-            return false;
+    if (!location->OwnedBy(m_id))
+        return false;
 
-        auto planet = std::dynamic_pointer_cast<const Planet>(location);
-        std::shared_ptr<const Ship> ship;
-        if (!planet)
-             return false;
+    if (!std::dynamic_pointer_cast<const Planet>(location))
+        return false;
 
+    if (build_type == BT_STOCKPILE) {
```

Comment:
The code is the way it is because of keeping the pattern in the other ProducibleItem functions. 
So if I change this - should I change the other Empire::ProducibleItem functions as well?

Reply:
Looking at it now / again, it seems fine as is